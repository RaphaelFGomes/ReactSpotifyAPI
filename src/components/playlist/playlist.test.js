import React from 'react';
import Playlist from './playlist';
import { shallow } from 'enzyme';

// Mounting - Full DOM rendering including child components
// Shallow - Renders only the single component, not including its children,
// this is useful to isolate the component for pure unit testing.
// Render - Renders to static HTML, including children

it('should render correctly with props', () => {
    const playlist = {
        "name":"Sweet Soul Chillout",
        "imageUrl":"https://pl.scdn.co/images/pl/default/dd52dbcce4e10a2feb519e9c282b4d3b0dcc7442",
        "songs":[
            {"name":"Valentine Love (Slow Version)","duration":238.158},
            {"name":"Valerie - Live At BBC Radio 1 Live Lounge, London / 2007","duration":233.733},
            {"name":"Just The Way You Are","duration":289.106}]};
    const component = shallow(<Playlist playlist={playlist}/>);
    expect(component).toMatchSnapshot();
});

