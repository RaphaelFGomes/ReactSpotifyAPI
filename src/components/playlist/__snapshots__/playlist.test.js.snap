// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly with props 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#808080",
      "color": "#000",
      "display": "inline-block",
      "fontFamily": "Papyrus",
      "padding": "10px",
      "width": "25%",
    }
  }
>
  <img
    alt="Image URL index: undefined"
    src="https://pl.scdn.co/images/pl/default/dd52dbcce4e10a2feb519e9c282b4d3b0dcc7442"
    style={
      Object {
        "width": "60px",
      }
    }
  />
  <h3
    style={
      Object {
        "fontWeight": "bold",
      }
    }
  >
    Sweet Soul Chillout
  </h3>
  <ul
    style={
      Object {
        "marginTop": "10px",
      }
    }
  >
    <li
      key="Valentine Love (Slow Version)"
      style={
        Object {
          "paddingTop": "2px",
        }
      }
    >
      Valentine Love (Slow Version)
    </li>
    <li
      key="Valerie - Live At BBC Radio 1 Live Lounge, London / 2007"
      style={
        Object {
          "paddingTop": "2px",
        }
      }
    >
      Valerie - Live At BBC Radio 1 Live Lounge, London / 2007
    </li>
    <li
      key="Just The Way You Are"
      style={
        Object {
          "paddingTop": "2px",
        }
      }
    >
      Just The Way You Are
    </li>
  </ul>
</div>
`;
